spec/models/iterations/project_iterations_spec.rb

```diff
@@ -18,6 +18,30 @@ module Iterations
       it 'returns a past iteration' do
         expect(subject.past_iterations).to all(be_a(PastIteration))
       end
+
+      context "when project didn't start in the same week day as the iterations" do
+        let(:story) { create(:story, :done, :with_project) }
+        let(:project) { story.project }
+
+        it 'all past iterations must start in the iteration_start_day' do
+          interation_start_day = 0
+          project.iteration_start_day = interation_start_day
+          project.start_date += interation_start_day + 1
+          project.start_date -= 3.weeks
+
+          past_iterations_start_date = subject.past_iterations.map{ |iteration| iteration.start_date.wday }
+
+          expect(past_iterations_start_date).to all(be project.iteration_start_day)
+        end
+
+        it 'considers the length of the first iteration to calculate the second iteration stories' do
```

Comment:
What do you think about adding more stories to that project (including some that should and shouldn't be at the second iteration) and use them at the assertions as well?